specs/gloas/beacon-chain.md

```diff
@@ -413,34 +472,51 @@ def is_parent_block_full(state: BeaconState) -> bool:
 
 ### Misc
 
-#### Modified `get_pending_balance_to_withdraw`
+#### New `convert_builder_index_to_validator_index`
+
+```python
+def convert_builder_index_to_validator_index(builder_index: BuilderIndex) -> ValidatorIndex:
+    return ValidatorIndex(builder_index | BUILDER_INDEX_FLAG)
+```
 
-*Note*: `get_pending_balance_to_withdraw` is modified to account for pending
-builder payments.
+#### New `convert_validator_index_to_builder_index`
 
 ```python
-def get_pending_balance_to_withdraw(state: BeaconState, validator_index: ValidatorIndex) -> Gwei:
-    return (
-        sum(
-            withdrawal.amount
-            for withdrawal in state.pending_partial_withdrawals
-            if withdrawal.validator_index == validator_index
-        )
-        # [New in Gloas:EIP7732]
-        + sum(
-            withdrawal.amount
-            for withdrawal in state.builder_pending_withdrawals
-            if withdrawal.builder_index == validator_index
-        )
-        # [New in Gloas:EIP7732]
-        + sum(
-            payment.withdrawal.amount
-            for payment in state.builder_pending_payments
-            if payment.withdrawal.builder_index == validator_index
-        )
+def convert_validator_index_to_builder_index(validator_index: ValidatorIndex) -> BuilderIndex:
+    return BuilderIndex(validator_index & ~BUILDER_INDEX_FLAG)
```

Comment:
I think more accurate bit mask would be `BuilderIndex(validator_index & (BUILDER_INDEX_FLAG - 1))` but it wouldn't matter as its value is not allowed to be bigger than 2**40.